src/property.ts

```diff
@@ -394,7 +394,7 @@ export class Property implements PropertyPath {
 
 				targetType.model.ready(() => {
 					let onChangeOf: PropertyPath[] = resolveDependsOn(this, "length", options.length.dependsOn);
-					if (isEntityType(this.propertyType)) {
+					if (this.isList) {
```

Comment:
_FIX_: This is a functional change. For some reason we were treating all "length" rules on entity-typed properties as list-length and string-length otherwise (assuming it was either `Entity[]` or `string`). So, you couldn't create a list-length rule on a property of type `string[]`. There was a test that was getting around this by declaring a property to be an entity list property, but actually using string values in reality.